src/Implementations/AzureTableStorage.cs

```diff
@@ -227,5 +230,64 @@ public async Task UpdateEntityAsync<T>(T oldEntity, T newEntity, string table) w
             TableOperation update = TableOperation.Replace(newEntity);
             await tableRef.ExecuteAsync(update, _options, null);
         }
+
+        /// <inheritdoc />
+        public async Task DeleteInBatchAsync(
+            string tableName,
+            DateTimeOffset cutOffDate,
+            CancellationToken cancellationToken = default(CancellationToken))
+        {
+            CloudTable table = await GetTableReferenceAsync(tableName);
+            TableQuery<DynamicTableEntity> query = new TableQuery<DynamicTableEntity>().Where(TableQuery.GenerateFilterConditionForDate("Timestamp", QueryComparisons.LessThan, cutOffDate));
+            IEnumerable<DynamicTableEntity> entities = await GetEntitiesAsync<DynamicTableEntity>(table, query, cancellationToken);
+            IEnumerable<TableBatchOperation> batches = GetBatchesForDelete(entities);
+            foreach (TableBatchOperation batch in batches)
+            {
+                await Task.WhenAll(table.ExecuteBatchAsync(batch));
+            }
+        }
+
+        private IEnumerable<TableBatchOperation> GetBatchesForDelete<T>(IEnumerable<T> tableEntities) where T : ITableEntity, new()
+        {
+            IList<TableBatchOperation> batches = new List<TableBatchOperation>();
+            IEnumerable<IGrouping<string, T>> partitionGroups = tableEntities.GroupBy(arg => arg.PartitionKey);
+
+            foreach (IGrouping<string, T> groupedEntities in partitionGroups)
+            {
+                int offset = 0;
+                IEnumerable<T> offsettedItems = groupedEntities.Skip(offset).Take(MAX_BATCH_SIZE);
```

Comment:
.Take will create a copy of the Elements in the Enumerable...is there a way to not use Take?

Reply:
Err...it will create a new IEnumerable<T> and copy MAX_BATCH_SIZE elements into the new IEnumerable<T> instead of doing it inline